### PR TITLE
feat!(db): database configuration for per-guild settings

### LINKED ISSUE #9: Enhancement: Add guild-specific settings
### Summary

Currently things like command prefix, help channel ID, etc. are hard coded.

To do anything useful past one guild we need to stand up a database schema that allows for settings thing per-guild.

### Basic Example

Here is a rough stub that needs worked on. I think (?) that we should introduce a new path under `src/` for database things specifically, so that they can be shared between the bot and web service.

```py
"""Shared models.

.. todo:: This is mostly a stub package that needs to be
    fleshed out.
"""

from sqlalchemy import BigInteger, Boolean, ForeignKey, String, Table
from sqlalchemy.orm import Mapped, mapped_column, relationship

from src.database.orm import TimestampedDatabaseModel

allowed_roles_table = Table(
    "allowed_roles",
    TimestampedDatabaseModel.metadata,
    mapped_column("github_config_id", ForeignKey("github_configs.id"), primary_key=True),
    mapped_column("role_id", BigInteger, primary_key=True),
)


class GuildConfig(TimestampedDatabaseModel):
    __tablename__ = "guild_configs"

    guild_id: Mapped[int] = mapped_column(BigInteger, primary_key=True)
    prefix: Mapped[str] = mapped_column(String(5), nullable=False, server_default="!", de

    help_channel_id: Mapped[int] = mapped_column(BigInteger, index=True)
    github_config: Mapped["GitHubConfig"] = relationship("GitHubConfig", uselist=False, back_populates="guild_config")


class GitHubConfig(TimestampedDatabaseModel):
    __tablename__ = "github_configs"

    guild_id: Mapped[int] = mapped_column(BigInteger, ForeignKey("guild_configs.guild_id"), nullable=False)
    discussion_sync: Mapped[bool] = mapped_column(Boolean, default=False)
    github_organization: Mapped[str] = mapped_column(String, nullable=True)
    github_repository: Mapped[str] = mapped_column(String, nullable=True)
    sync_label: Mapped[str] = mapped_column(String, nullable=True)
    issue_linking: Mapped[bool] = mapped_column(Boolean, default=False)
    comment_linking: Mapped[bool] = mapped_column(Boolean, default=False)
    pep_linking: Mapped[bool] = mapped_column(Boolean, default=False)

    allowed_roles: Mapped[list[int]] = relationship(
        "Role",
        secondary=allowed_roles_table,
        back_populates="github_configs",
    )

    guild_config: Mapped[GuildConfig] = relationship("GuildConfig", back_populates="github_config")


class Role(TimestampedDatabaseModel):
    __tablename__ = "roles"

    id: Mapped[int] = mapped_column(BigInteger, primary_key=True)

    github_configs: Mapped[list[GitHubConfig]] = relationship(
        "GitHubConfig",
        secondary=allowed_roles_table,
        back_populates="allowed_roles",
    )
```

### Drawbacks and Impact

[Complexity.](https://grugbrain.dev/#grug-on-complexity)

### Unresolved questions

- Are there any configurable things we would like to add on top of the above example?

### PR DESCRIPTION
**Pull request by JacobCoffee**
[//]: # "By submitting this pull request, you agree to:"
[//]: # "- follow [PSFs's Code of Conduct](https://www.python.org/psf/conduct/)"
[//]: # "- follow [Bytes's contribution guidelines](https://github.com/JacobCoffee/byte/blob/main/CONTRIBUTING.rst)"

### Pull Request Checklist

- [ ] New code has 100% test coverage
- [ ] (If applicable) The prose documentation has been updated to reflect the changes introduced by this PR
- [ ] (If applicable) The reference documentation has been updated to reflect the changes introduced by this PR
- [ ] Pre-Commit Checks were ran and passed
- [ ] Tests were ran and passed

### Description
[//]: # "Please describe your pull request for new release changelog purposes"

- initial db setup for #9

### Close Issue(s)
[//]: # "Please add in issue numbers this pull request will close, if applicable"
[//]: # "Examples: Fixes #4321 or Closes #1234"

- Closes #9

### CODE REVIEW

**Comment by JacobCoffee** [comment]
Migrations to not autogenerate currently for some reason

File: src/server/lib/db/utils.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
+"""Database utilities."""
```
Comment (cofin) [inline src/server/lib/db/utils.py:1]: I don't think you need this file any more

File: src/server/domain/db/models.py [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+
+    Here, a guild should be able to configure their own GitHub organization, StackOverflow tags, etc.
+    """
+
+    __tablename__ = "guild_config"
+    __table_args__ = {"comment": "Configuration for a Discord guild."}
+
+    guild_id: Mapped[int] = mapped_column(BigInteger, unique=True, index=True)
+    guild_name: Mapped[str] = mapped_column(String(100))
+    prefix: Mapped[str] = mapped_column(String(5), nullable=False, server_default="!", default="!")
```
Comment (Alc-Alc) [inline src/server/domain/db/models.py:31]: nit

File: src/server/domain/db/models.py [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
```
Comment (Alc-Alc) [inline src/server/domain/db/models.py:31]:
```suggestion
    prefix: Mapped[str] = mapped_column(String(5), server_default="!", default="!")
```

**Pull request merged**